docs/generating-tests.rst

```diff
@@ -0,0 +1,370 @@
+
+
+``env`` field contains some parameters in a straightforward way.
+
+``pre`` field describes the pre-state account-wise:
+
+.. code::
+
+     "pre" : {
+        "0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6" : {
+            "balance" : "0x0de0b6b3a7640000",
+            "code" : "{ (MSTORE 0 0x112233445566778899aabbccddeeff) (RETURNDATACOPY 0 0 32) (SSTORE 0 (MLOAD 0)) }",
+            "code" : "0x306000526020600060003e600051600055",
+            "nonce" : "0x00",
+            "storage" : {
+                "0x00" : "0x01"
+            }
+        }
+     }
+
+
+As specified in the Yellow Paper, an account contains a balance, a code, a nonce and a storage.
+
+Notice the ``code`` field is duplicated.  If many fields exist under the same name, the last one is used. In this particular case, the LLL compiler was not ready to parse the new instruction ``RETURNDATACOPY`` so a compiled runtime bytecode is added as the second ``code`` field [#]_.
+
+.. [#] Unless you are testing malformed bytecode, always try to keep the LLL code in the test filler.  LLL code is easier to understand and to modify.
+
+
+This particular test expected to see ``0`` in the first slot in the storage. In order to make this change visible, the pre-state has ``1`` there.
+
+Usually, there is another account that acts as the initial caller of the transaction.
+
+``transaction`` field is somehow interesting because it can describe a multidimensional array of test cases.  Notice that ``data``, ``gasLimit`` and ``value`` fields are lists.
+
+.. code::
+
+   "transaction" : {
+        "data" : [
+            "", "0xaaaa", "0xbbbb"
+        ],
+        "gasLimit" : [
+            "0x0a00000000",
+            "0x0"
+        ],
+        "gasPrice" : "0x01",
+        "nonce" : "0x00",
+        "secretKey" : "0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8",
+        "to" : "0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6",
+        "value" : [
+            "0x00"
+        ]
+    }
+
+
+Since ``data`` has three elements and ``gasLimit`` has two elements, the above ``transaction`` field specifies six different transactions.  Later, in the ``expect`` section, ``data : 1`` would mean the ``0xaaaa`` as data, and ``gasLimit : 0`` would mean ``0x0a00000000`` as gas limit.
+
+Moreover, these transactions are tested under different versions of the protocol.
+
+``expect`` field of the filler specifies the expected fields of the state after the transaction.  The ``expect`` field does not need to specify a state completely, but it should specify some features of some accounts.  ``expect`` field is a list. Each element talks about some elements of the multi-dimensional array defined in ``transaction`` field.
+
+.. code::
+
+   "expect" : [
+        {
+            "indexes" : {
+                "data" : 0,
+                "gas" : -1,
+                "value" : -1
+            },
+            "network" : ["Frontier", "Homestead"],
```

Comment:
network names / forks been extended to format  

`>=Homestead`

the order of forks:   ` Frontier < Homestead < EIP150 < EIP158 < Byzantium < Constantinople `

Reply:
Ok, have done additions to ``network`` format explanation and an extra note on fork order.